ios/browser/api/net/certificate_utility.mm

```diff
@@ -0,0 +1,143 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#import "brave/ios/browser/api/net/certificate_utility.h"
+
+#include <algorithm>
+#include <memory>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "base/base64.h"
+#include "base/bind.h"
+#include "base/logging.h"
+#include "base/memory/ref_counted.h"
+#include "base/strings/sys_string_conversions.h"
+
+#include "net/cert/internal/cert_errors.h"
+#include "net/cert/internal/parsed_certificate.h"
+#include "net/cert/pem.h"
+#include "net/cert/x509_certificate.h"
+#include "net/cert/x509_util.h"
+#include "net/der/input.h"
+#include "net/http/hsts_info.h"
+#include "net/http/transport_security_state.h"
+#include "net/net_buildflags.h"
+#include "net/tools/transport_security_state_generator/cert_util.h"
+#include "net/tools/transport_security_state_generator/spki_hash.h"
+
+#include "third_party/boringssl/src/include/openssl/mem.h"
+#include "third_party/boringssl/src/include/openssl/sha.h"
+#include "third_party/boringssl/src/include/openssl/x509.h"
+
+namespace net {
+namespace {
+#if BUILDFLAG(INCLUDE_TRANSPORT_SECURITY_STATE_PRELOAD_LIST)
+#include "net/http/transport_security_state_static.h"  // nogncheck
+#endif  // INCLUDE_TRANSPORT_SECURITY_STATE_PRELOAD_LIST
+}
+}  // net
+
+namespace net {
+// from: net/tools/transport_security_state_generator/cert_util.h
```

Comment:
Why don't we use it directly then instead of redefining it?

Reply:
Can't link against this file for sure. I have tried it for hours it will compile but it will not LINK. So I can include the file, everything compiles great, but then it doesn't link because this is a tool that is just ran and generates the keys from certs, then some Chromium code includes this file. It is available to android and desktop, but not Chromium iOS or Brave iOS.